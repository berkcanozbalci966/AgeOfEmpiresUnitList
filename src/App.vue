<template>
  <div class="container">
    <div class="nav-container">
      <span></span>
      <h1>{{ $router.currentRoute.value.name }}</h1>

      <div id="nav">
        <router-link :class="currentRoute === 'Home' ? 'active' : null" to="/"
          >Home</router-link
        >
        <router-link
          :class="currentRoute === 'Units' ? 'active' : null"
          to="/units"
          >Units</router-link
        >
      </div>
    </div>
    <router-view />
  </div>
</template>
<script>
export default {
  computed: {
    currentRoute() {
      return this.$router.currentRoute.value.name;
    },
  },
};
</script>

<style lang="scss" scoped>
.nav-container {
  justify-content: space-between;
  align-items: center;
  display: flex;
  margin-bottom: 40px;
  #nav {
    a {
      color: #125d98;
      text-decoration: none;
      margin-right: 5px;
      font-size: 18px;
      letter-spacing: 0.5px;
      &.active {
        font-weight: 700;
      }
      &:last-child {
        margin-right: 0;
      }
    }
  }
  h1 {
    color: #125d98;
    letter-spacing: 0.5;
  }
}
</style>
