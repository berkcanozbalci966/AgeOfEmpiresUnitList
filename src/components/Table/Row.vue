<template>
  <tr>
    <td>{{ unitInfo.id }}</td>
    <td>{{ unitInfo.name }}</td>
    <td>{{ unitInfo.age }}</td>
    <td>{{ unitInfo.cost }}</td>
  </tr>
</template>

<script>
export default {
  props: {
    unitInfo: {
      type: Object,
      default: () => ({}),
    },
  },
  mounted() {
    console.log(this.unitInfo);
  },
};
</script>

<style></style>
