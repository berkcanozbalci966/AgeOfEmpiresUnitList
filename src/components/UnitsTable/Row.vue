<template>
  <tr data-test="unitsrow" @click="routerNavigate()">
    <td>{{ unitInfo.id }}</td>
    <td>{{ unitInfo.name }}</td>
    <td>{{ unitInfo.age }}</td>
    <td>{{ unitInfo.cost }}</td>
  </tr>
</template>

<script>
export default {
  props: {
    unitInfo: {
      type: Object,
      default: () => ({}),
    },
  },
  mounted() {},
  methods: {
    routerNavigate() {
      this.$router.push({
        name: "Unit Detail",
        params: { id: this.unitInfo.id },
      });
    },
  },
};
</script>

<style lang="scss" scoped>
tr {
  &:hover {
    cursor: pointer;
    background-color: "red";
  }
}
</style>
