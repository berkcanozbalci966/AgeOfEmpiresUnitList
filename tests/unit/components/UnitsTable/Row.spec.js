import { shallowMount, mount, flushPromises } from "@vue/test-utils";
import Vuex from "vuex";
import router from "../../../../src/router/index";
import Row from "../../../../src/components/UnitsTable/Row";
import UnitsModule from "../../../../src/store/modules/units";
import AgeModule from "../../../../src/store/modules/age";
import App from "../../../../src/App";

describe("Row", () => {
  let store;

  beforeEach(() => {
    store = new Vuex.Store({
      modules: {
        Units: UnitsModule,
        Age: AgeModule,
      },
    });
  });

  it("Render correctly", () => {
    const component = shallowMount(Row, {
      props: {
        unitInfo: {
          id: 1,
          name: "Test",
          age: "Dark",
          cost: "50 food 50 gold",
        },
      },
    });
    expect(component.html()).toMatchSnapshot();
  });
  // Router Test
  it("Row click after router trigger correctly", async () => {
    router.push("/units");
    await router.isReady();
    const app = mount(App, {
      global: {
        plugins: [router, store],
      },
    });
    await app.find("[data-test='unitsrow']").trigger("click");
    await flushPromises();
    expect(app.find("h1").text()).toBe("Unit Detail");
  });
});
