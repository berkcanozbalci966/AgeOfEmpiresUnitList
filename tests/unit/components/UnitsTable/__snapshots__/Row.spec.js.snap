// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row Render correctly 1`] = `
<tr data-test="unitsrow">
  <td>1</td>
  <td>Test</td>
  <td>Dark</td>
  <td>50 food 50 gold</td>
</tr>
`;
